### DISCUSSION TITLE
Run job before or after helm release is upgraded

### DISCUSSION THREAD

**Discussion by onedr0p** [fluxcd/flux2#2442] [Proposals]
I am thinking that something like the [Resource Hooks in ArgoCD](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks/) would be beneficial.

I've seen a couple of methods described on how to attempt to do this already but there's going to be a situation where:

- You don't control the development of a helm chart so you can't use the helm hooks, it's maintained outside your org and you would like to run a `Job` before upgrading the Helm chart (e.g. snapshot a persistent volume.)

- In regards to an `initContainer`, `k8s-wait-for` and a `Job`, you do not want to run an `Job` on a restart of the pod, only an Helm chart upgrade. There's no way I can think of how that would know the difference between a `HelmRelease` update or a restart of a `Pod`.

**Comment by xunholy** [comment]
I would also like to be able to have hooks for remediation events, I have scenarios where certain application won't be able to gracefully rollback or be remediated. Ideally I'd also be able to trigger other remediations events for stateful applications.

**Comment by kingdonb** [comment]
These are great ideas to bring to the community meeting, so we can get the consensus of the group about them.

Thursday at 11:00AM Eastern (tomorrow) is the Late Community Meeting, we will add it to the agenda and try to have a discussion about the idea. You're welcome to join and help us understand the specific goals, since it sounds like you know exactly what you're after from this feature idea!

Thanks for bringing this up.

**Reply by onedr0p** [reply]
Wish I could join but I'll be busy, sorry! Let me know if you need further context or if the problem can be solved another way. I'll be sure to watch the video later this week.

**Comment by sebastiangaiser** [comment]
Are there any news about that? 
Is there another way to push this feature except the Community Meeting?

**Reply by kingdonb** [reply]
Helm has its own lifecycle hooks, can you say a bit more specifically what you are looking for? I don't think we want to create a second version of helm's lifecycle hooks, rather Flux should lean into the existing support for helm hooks which is established and well worn in the wider ecosystem of Kubernetes.

Argo hooks are subtly incompatible with Helm hooks, though they map the hooks from Helm to Argo anyway, and it causes people issues. The most effective way to get the changes you want into Flux is to bring them to the dev meetings. We have 2-3x dev meetings per week staggered at different times, if you can make it, we can take your feedback and possibly help you solve it without changing Flux or adding a new (third standard) there's an XKCD for this!
